components/clp-py-utils/clp_py_utils/initialize-results-cache.py

```diff
@@ -0,0 +1,146 @@
+import argparse
+import logging
+import sys
+from urllib.parse import urlparse
+
+from pymongo import IndexModel, MongoClient
+from pymongo.errors import OperationFailure
+
+# Setup logging
+# Create logger
+logger = logging.getLogger(__file__)
+logger.setLevel(logging.INFO)
+# Setup console logging
+logging_console_handler = logging.StreamHandler()
+logging_formatter = logging.Formatter("%(asctime)s [%(levelname)s] %(message)s")
+logging_console_handler.setFormatter(logging_formatter)
+logger.addHandler(logging_console_handler)
+
+
+def check_replica_set_status(client: MongoClient, netloc: str) -> tuple[bool, bool]:
+    """
+    Checks the current replica set status of the MongoDB server and determines whether it needs to
+    be configured or reconfigured.
+
+    :param client:
+    :param netloc: The network location for which the replica set configuration is being checked.
```

Comment:
How do you think about?

The network location of the replica set configuration

It's simpler but not sure if it's accurate enough.

Reply:
sure
```suggestion
    :param netloc: The network location of the replica set configuration.
```